core/templates/pages/library-page/library-page.component.ts

```diff
@@ -213,7 +217,35 @@ export class LibraryPageComponent {
     return this.urlInterpolationService.getStaticImageUrl(imagePath);
   }
 
+  getCorrespondingContainerIDForButtonClick(buttonID: string): string {
+    let containerIDValue = buttonID.split('-')[3];
+    return '#card-container-' + containerIDValue;
+  }
+
+  toggleButtonText(container: HTMLElement, button: HTMLElement): void {
+    if (container.classList.contains('limit-cards-shown')) {
+      button.textContent = 'Collapse Section';
+    } else {
+      button.textContent = 'See More';
+    }
+  }
+
+  toggleCardContainerHeightInMobileView(click: Event): void {
+    let buttonElement = click.target as HTMLElement;
+    let containerID = this.getCorrespondingContainerIDForButtonClick(
+      buttonElement.id);
+    let containerElement = (
+      document.querySelector(containerID) as HTMLElement);
+
+    this.toggleButtonText(containerElement, buttonElement);
+    containerElement.classList.toggle('limit-cards-shown');
```

Comment:
Is there a way to achieve the same directly in the template (e.g. using ngClass or similar)? We generally try to avoid manipulating the DOM directly because Angular may not be aware of it, this can cause unexpected side effects. Also, it would be great to avoid the use of HTML "id" attributes.

Reply:
In addition to the 'classList.toggle()' method, I'm assuming you are referring to the 'toggleButtonText()' method as well, which manipulates the textContent property of the button elements.

One alternative I could think of is to have an array (say, 'mobileLibraryGroupsProperties'), where each element is an object with two properties -- a boolean (say 'inExpandedState'), and a string (say 'buttonText'). All the booleans will be initialized as false, and the strings will be initialized as "See More".

On each of the card containers, I could then have:
```
[ngClass]="{'limit-cards-shown: !mobileLibraryGroupsProperties[idx].inExpandedState'}" 
```
and within each of the button elements, I could have

```
{{ mobileLibraryGroupsProperties[idx].buttonText }}
```
as the text.

Every time a toggle button is clicked, the corresponding idx value (which is already available in the template) is passed to a handler function, which

- flips the value of inExpandedState at the corresponding index
- changes the buttonText value to "Collapse Section" (or vice-versa) at the corresponding index

Of course, this would also eliminate the need of the 'id' attribute.
Do you think this is an ideal alternative?

Reply:
Yup, sounds good, thanks for explaining! One suggestion: In the future, feel free to push the changes, it speeds up the review and the approach is clearer, we can always revert back if something needs to change.

Reply:
> push the changes, it speeds up the review and the approach is clearer

Got it, will do so from now on.